### PR TITLE
[website] Use new designs with tooltips for allowance toggles

### PR DESCRIPTION
**Pull request by fragosti**
## Description

screen shot 2018-07-24 at 3 55 42 pm

https://0x.wake.com/96

The circular progress spinners are pretty annoying as they wobble. I made a new `Spinner` component that animates more or less like the material UI one but isn't on meth.

## Testing instructions

https://dogfood.0xproject.com/ and play around!

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist:

*   [ ] Prefixed the title of this PR with `[WIP]` if it is a work in progress.
*   [x] Prefixed the title of this PR with bracketed package name(s) corresponding to the changed package(s). For example: `[sol-cov] Fixed bug`.
*   [ ] Added tests to cover my changes, or decided that tests would be too impractical.
*   [ ] Updated documentation, or decided that no doc change is needed.
*   [ ] Added new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: packages/website/ts/components/inputs/allowance_state_toggle.tsx [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    }
+    constructor(props: AllowanceStateToggleProps) {
+        super(props);
+        const tokenState = props.tokenState;
+        this.state = {
+            allowanceState: AllowanceStateToggle._getAllowanceState(tokenState),
+            prevTokenState: tokenState,
+        };
+    }
+
```
Comment (BMillman19) [inline packages/website/ts/components/inputs/allowance_state_toggle.tsx]: nit: extra newline here

File: packages/website/ts/components/inputs/allowance_state_toggle.tsx [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+            });
+        }
+    }
+    private _getTooltipContent(): React.ReactNode {
+        const symbol = this.props.token.symbol;
+        const isLockingToken = this.props.tokenState.allowance.gt(0);
+        switch (this.state.allowanceState) {
+            case AllowanceState.Loading:
+                return (
+                    <Text noWrap={true} fontColor="white">
```
Comment (BMillman19) [inline packages/website/ts/components/inputs/allowance_state_toggle.tsx]: nit: `colors.white` for consistency

File: packages/website/ts/components/inputs/allowance_state_toggle.tsx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+            const tokenState = nextProps.tokenState;
+            this.setState({
+                prevTokenState: tokenState,
+                allowanceState: AllowanceStateToggle._getAllowanceState(nextTokenState),
+            });
+        }
+    }
+    private _getTooltipContent(): React.ReactNode {
+        const symbol = this.props.token.symbol;
+        const isLockingToken = this.props.tokenState.allowance.gt(0);
```
Comment (BMillman19) [inline packages/website/ts/components/inputs/allowance_state_toggle.tsx]: can we put this in the Loading case or AND it with `this.state.allowanceState === AllowanceState.Loading`

Comment (BMillman19) [reply packages/website/ts/components/inputs/allowance_state_toggle.tsx]: I think we need to handle the case when the tokenState is loading its initial state, when we are doing the initial load the tooltip incorrectly displays "locking/unlocking X" when really we just are waiting to read the state from the blockchain

File: packages/website/ts/components/ui/allowance_state_view.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+                <Container position="relative" top="3px" left="5px">
+                    <Spinner size={18} strokeSize={2} />
+                </Container>
+            );
+        default:
+            return null;
+    }
+};
+
+const renderCheck = (color: string = '#37D400') => (
```
Comment (BMillman19) [inline packages/website/ts/components/ui/allowance_state_view.tsx]: can we put this color in the shared colors file?

File: packages/website/ts/components/inputs/allowance_state_toggle.tsx (hunk 1 above)
Comment (fragosti) [reply packages/website/ts/components/inputs/allowance_state_toggle.tsx]: this is a tslint rule :P

**Pull request merged**